lib/backend/lite/lite.go

```diff
@@ -887,6 +785,92 @@ func (l *Backend) DeleteRange(ctx context.Context, startKey, endKey []byte) erro
 	})
 }
 
+func (l *Backend) ConditionalUpdate(ctx context.Context, i backend.Item) (*backend.Lease, error) {
+	if i.Key == nil {
+		return nil, trace.BadParameter("missing parameter key")
+	}
+
+	rev := backend.CreateRevision()
+	err := l.inTransaction(ctx, func(tx *sql.Tx) error {
+		now := l.clock.Now().UTC()
+
+		stmt, err := tx.PrepareContext(ctx, "UPDATE kv SET value = ?, expires = ?, modified = ?, revision = ? WHERE key = ? AND revision = ?")
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer stmt.Close()
+
+		result, err := stmt.ExecContext(ctx, i.Value, expires(i.Expires), id(now), rev, string(i.Key), i.Revision)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		rows, err := result.RowsAffected()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if rows == 0 {
+			return trace.Wrap(backend.ErrIncorrectRevision)
+		}
+		if !l.EventsOff {
+			stmt, err = tx.PrepareContext(ctx, "INSERT INTO events(type, created, kv_key, kv_modified, kv_expires, kv_value) values(?, ?, ?, ?, ?, ?)")
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			defer stmt.Close()
+
+			if _, err := stmt.ExecContext(ctx, types.OpPut, now, string(i.Key), id(now), expires(i.Expires), i.Value); err != nil {
+				return trace.Wrap(err)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	i.Revision = rev
+	return backend.NewLease(i), nil
+}
+
+func (l *Backend) ConditionalDelete(ctx context.Context, key []byte, revision string) error {
+	if len(key) == 0 {
+		return trace.BadParameter("missing parameter key")
+	}
+
+	return l.inTransaction(ctx, func(tx *sql.Tx) error {
+		stmt, err := tx.PrepareContext(ctx, "DELETE FROM kv WHERE key = ? AND revision = ?")
```

Comment:
I don't know how to feel about returning no error when deleting an expired item, especially with a conditional operation. The memory backend that was just merged will error out in that case, however.